targets/TARGET_Infineon/TARGET_XMC4XXX/i2c_api.c

```diff
@@ -0,0 +1,347 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2015 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/***************************************************************** Includes **/
+#include "pinmap.h"
+#include "PeripheralPins.h"
+#include "xmc4_helper.h"
+#include "mbed_assert.h"
+#include "i2c_api.h"
+#include "xmc_gpio.h"
+#include "xmc_i2c.h"
+
+/****************************************************************** Defines **/
+#define I2C_TIMEOUT 100000
+
+#define I2C_START_SEND      0
+#define I2C_STOP_SEND       1
+
+/******************************************************************* Macros **/
+
+/****************************************************************** Globals **/
+
+/* i2c configuration */
+XMC_I2C_CH_CONFIG_t i2c_cfg =
+{
+        .baudrate = 50000U, // Default to 100kHz
+};
+
+/* pin configuration for i2c sda */
+XMC_GPIO_CONFIG_t i2c_sda_pin_cfg =
+{
+        .output_strength = XMC_GPIO_OUTPUT_STRENGTH_MEDIUM
+};
+
+/* pin configuration for i2c sdc */
+XMC_GPIO_CONFIG_t i2c_scl_pin_cfg =
+{
+        .output_strength = XMC_GPIO_OUTPUT_STRENGTH_MEDIUM
```

Comment:
8 spaces instead of 4?

see https://os.mbed.com/docs/v5.8/reference/style.html